xlml/utils/metric.py

```diff
@@ -267,6 +285,73 @@ def process_tensorboard_summary(
   return [metric_history_rows], [metadata_history_rows]
 
 
+def get_run_history_rows(
+    base_id: str,
+    aggregated_metrics: Dict[str, Any],
+    metadata: Dict[str, Any],
+    task_test_config: test_config.TestConfig[test_config.Tpu],
+) -> List[List[bigquery.RunHistoryRow]]:
+  """Create run_history rows to insert into BigQuery
+
+  Args:
+    base_id: The unique ID for this test job.
+    aggregated_metrics: Dict of aggregated metrics
+    metadata: Dict of metadata
+    task_test_config: Test config
+
+  Returns:
+    A list of RunHistoryRow for a test run in a test job.
+  """
+  uuid = generate_row_uuid(base_id, 0)
+  dcn_data_parallelism = metadata["dcn_data_parallelism/text_summary"]
+  dcn_fsdp_parallelism = metadata["dcn_fsdp_parallelism/text_summary"]
+  dcn_sequence_parallelism = metadata["dcn_sequence_parallelism/text_summary"]
+  dcn_tensor_parallelism = metadata["dcn_tensor_parallelism/text_summary"]
+  dcn_autoregressive_parallelism = metadata[
+      "dcn_autoregressive_parallelism/text_summary"
+  ]
+  ici_data_parallelism = metadata["ici_data_parallelism/text_summary"]
+  ici_fsdp_parallelism = metadata["ici_fsdp_parallelism/text_summary"]
+  ici_sequence_parallelism = metadata["ici_sequence_parallelism/text_summary"]
+  ici_tensor_parallelism = metadata["ici_tensor_parallelism/text_summary"]
+  ici_autoregressive_parallelism = metadata[
+      "ici_autoregressive_parallelism/text_summary"
+  ]
+  throughput = (
+      int(metadata["global_batch_size_to_train_on/text_summary"])
+      * int(metadata["max_target_length/text_summary"])
+  ) / aggregated_metrics["perf/step_time_seconds"]
+  precision = (
+      "bfloat16"
+      if not metadata["quantization/text_summary"]
+      else metadata["quantization/text_summary"]
+  )
+
+  run_history_row = bigquery.RunHistoryRow(
+      uuid=uuid,
+      description=task_test_config.test_name,
+      platform="Cloud",
+      date=datetime.datetime.now(),
+      base_cl="",
```

Comment:
What's the plan for this metadata?

Reply:
I'm not quite sure yet, I think I will need to add a change in MaxText to get this value into the tensorboard file. But for now will leave it blank.